langchain/vectorstores/qdrant.py

```diff
@@ -454,18 +466,20 @@ def max_marginal_relevance_search(
         Returns:
             List of Documents selected by maximal marginal relevance.
         """
+        query_vector = self._embed_query(query)
+        if self.vector_name is not None:
+            query_vector = (self.vector_name, query_vector)  # type: ignore[assignment]
 
-        embedding = self._embed_query(query)
         results = self.client.search(
             collection_name=self.collection_name,
-            query_vector=embedding,
+            query_vector=query_vector,
             with_payload=True,
             with_vectors=True,
             limit=fetch_k,
         )
         embeddings = [result.vector for result in results]
         mmr_selected = maximal_marginal_relevance(
-            np.array(embedding), embeddings, k=k, lambda_mult=lambda_mult
+            np.array(query_vector), embeddings, k=k, lambda_mult=lambda_mult
```

Comment:
we probably don't want to pass in vector_name to this, right?

Reply:
@dev2049 Thanks for pointing this out! This was not covered well in the tests, so I extended and fixed them. I would appreciate another look!